deviations/views.py

```diff
@@ -1,14 +1,59 @@
 from django.contrib import messages
+from django import forms
 from django.db import IntegrityError
 from django.shortcuts import get_object_or_404
-from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import ugettext_lazy as _, ngettext
+from userprofile.models import UserProfile
 
 from course.viewbase import CourseInstanceBaseView, CourseInstanceMixin
 from lib.viewbase import BaseFormView, BaseRedirectView
 from authorization.permissions import ACCESS
-from .forms import DeadlineRuleDeviationForm
+from .forms import DeadlineRuleDeviationForm, RemoveDeadlineRuleDeviationForm
 from .models import DeadlineRuleDeviation
 from exercise.models import BaseExercise
+from itertools import chain
+
+
+class OverrideDeadlinesView(CourseInstanceMixin, BaseFormView):
+    access_mode = ACCESS.TEACHER
+    template_name = "deviations/override.html"
+    form_class = forms.Form
+
+    def get_success_url(self):
+        return self.instance.get_url("deviations-list-dl")
+
+    def get_common_objects(self):
+        super().get_common_objects()
+        deviation_list = []
+        for e, s in self.request.session['already_have_deviation']:
+            try:
+                deviation_list.append(DeadlineRuleDeviation.objects.get(
+                        exercise=BaseExercise.objects.get(id=e),
+                        submitter=UserProfile.objects.get(id=s),
+                        exercise__course_module__course_instance=self.instance))
+            except:
+                pass
```

Comment:
@Mankro how should we handle this exception?

Reply:
You must name the exception, something like `DeadlineRuleDeviation.DoesNotExist`.

Reply:
The user tries to add deviations here, so let's create the deviations since they don't exist for these students (even though they existed before this view was loaded).